src/Microsoft.Extensions.Diagnostics.HealthChecks.Abstractions/HealthCheckResult.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.Diagnostics.HealthChecks
+{
+    /// <summary>
+    /// Represents the result of a health check.
+    /// </summary>
+    public struct HealthCheckResult
```

Comment:
Should this expose the source name?

Reply:
Yeah, I thought about that. The somewhat weird thing is that this object is produced by the `IHealthCheck` itself so it seems awkward to have it able to set the name of the result (potentially to something different from it's actual name). The consumer of `CheckHealthAsync` always knows which result maps to which check (because either they called the `IHealthCheck` manually and have it's `Name` available, or via `CompositeHealthCheckResult`'s dictionary) so I don't think it's necessary to have the name here.